starfish/core/imagestack/imagestack.py

```diff
@@ -1147,12 +1148,18 @@ def max_proj(self, *dims: Axes) -> "ImageStack":
         -------
         np.ndarray :
             max projection
-
         """
         max_projection = self._data.max([dim.value for dim in dims])
         max_projection = max_projection.expand_dims(tuple(dim.value for dim in dims))
         max_projection = max_projection.transpose(*self.xarray.dims)
-        max_proj_stack = self.from_numpy(max_projection.values)
+        physical_coords = self.xarray.coords
+        if Axes.ZPLANE in dims:
+            # if we max proj by z, take average coord
+            physical_coords[Coordinates.Z].values[0] = \
```

Comment:
don't calculate the average twice.

also, the physical_coords[Coordinates.Z] is the wrong size!  It should be size = 2, according to the contract for `from_numpy` (min, max).

furthermore, the size of the coords for X/Y is also wrong.  It's currently the size of the X/Y, and presumably it ought to also be 2. 

```
average_z_coordinate = np.average((self.xarray.coords[Coordinates.Z])
physical_coords[Coordinates.Z] = xr.DataArray(...)
```

Furthermore, if one max projects across another dimension where we assign coordinates (like x or y), we would presumably need to do the same thing.  Alternatively, we could explicitly disallow max projecting across x or y, but if we want to do that, we should clear it with Deep/Ambrose/Kevin/Brian.

Reply:
There is no min/max contract in from_numpy, it takes the first and last values from the array. So they can be any size

Reply:
But I can do that here if you think that's cleaner